layout_display/layout_display.py

```diff
@@ -1,125 +1,156 @@
+        # If something unrelated changes like a new dictionary
+        # being added then the system name will not be in config
+        if 'system_name' not in config:
+            return
 
-    def drawText(self, event, qp):
-        qp.setPen(QColor(168, 34, 3))
-        qp.setFont(QFont('Decorative', 10))
-        qp.drawRect(10, 10, 100, 100)
-        qp.drawText(event.rect(), Qt.AlignCenter, 'booop')
+        self._stroke = []
+        self._numbers = set(system.NUMBERS.values())
+        self._numbers_to_keys = {v: k for k, v in system.NUMBERS.items()}
+        self._number_key = system.NUMBER_KEY
+        self._system_name = config['system_name']
+
+        # If the user has no valid preference then fall back to the default
+        preferred_layout_file = self.get_preferred_layout(self._system_name)
+        if preferred_layout_file and self._layout.load_from_file(preferred_layout_file):
+            self.label_layout_name.setText(self._layout.name)
+            self.layout_display_view.initialize_view(self._layout)
+        else:
+            self.on_reset()
+
+    def on_stroke(self, stroke: Stroke):
+        ''' Updates state based off of the latest stroke by the user '''
 
-    def on_stroke(self, stroke):
         keys = stroke.steno_keys[:]
+
+        # Handle converting numbers in the stroke to the actual key values
         if any(key in self._numbers for key in keys):
-            keys.append('#')
+            keys.append(self._number_key)
         keys = [self._numbers_to_keys[x] if x in self._numbers_to_keys else x for x in keys]
+
         self._stroke = keys
-        self.update()
+        self.layout_display_view.update_view(self._layout, keys)
+
+    def on_reset(self):
+        ''' Resets the layout to the built-in default layout '''
+
+        self._remove_system_file_map(self._system_name)
+
+        self._layout.load_from_resource(':/layout_display/english_stenotype.json')
+        self.label_layout_name.setText(self._layout.name)
+        self.layout_display_view.initialize_view(self._layout)
+
+    def on_load(self):
+        ''' Gets a layout file from the user to load '''
+
+        # The API says this should return a string, but it returns a tuple
+
+        file_path, _filter = QFileDialog.getOpenFileName(self, _('Open Layout File'),
+                                                         CONFIG_DIR, '(*.json)')
+
+        # If the user cancelled out of the dialog then we will have a null string
+        if not file_path:
+            return
+
+        if self._layout.load_from_file(file_path):
+            self._add_system_file_map(self._system_name, file_path)
+            self.label_layout_name.setText(self._layout.name)
+            self.layout_display_view.initialize_view(self._layout)
+        else:
+            self.on_reset()
+
+    def get_preferred_layout(self, system_name: str) -> str:
+        ''' Gets the user's preferred layout file for the given system '''
+
+        file_path = ''
```

Comment:
This pattern can be simplified with

     file_path = self._system_file_map.get(system_name)

This will get set to None if not found. I'd also say use None instead of '', but you can get '' when there isn't a key with: self._system_file_map.get(system_name, '')